src/mame/misc/orientp.cpp

```diff
@@ -0,0 +1,179 @@
+u39 at89s51 second mcu for protection.
+
+// TODO:
+Need hardware info.
+Hook up nvram inputs opll and adpcm.
+east8 Has undumped mcu and adpcm rom.
+Need Layout as and Add segment display as marywu.cpp
+*/
+
+#include "emu.h"
+
+#include "cpu/mcs51/mcs51.h"
+#include "machine/i8255.h"
+#include "machine/i8279.h"
+#include "sound/ay8910.h"
+#include "sound/ymopl.h"
+#include "sound/okim6295.h"
+#include "speaker.h"
+
+namespace {
+
+class orientalpearl_state : public driver_device
+{
+public:
+	orientalpearl_state(const machine_config &mconfig, device_type type, const char *tag)
+		: driver_device(mconfig, type, tag)
+	{ }
+
+	void orientp(machine_config &config);
+
+private:
+	void io_map(address_map &map);
+	void program_map(address_map &map);
+
+protected:
+	virtual void machine_start() override;
+
+};
+
+static INPUT_PORTS_START( orientp )
+	PORT_START("IN0")
+	PORT_BIT( 0x01, IP_ACTIVE_LOW, IPT_UNKNOWN )
+	PORT_BIT( 0x02, IP_ACTIVE_LOW, IPT_UNKNOWN )
+	PORT_BIT( 0x04, IP_ACTIVE_LOW, IPT_UNKNOWN )
+	PORT_BIT( 0x08, IP_ACTIVE_LOW, IPT_UNKNOWN )
+	PORT_BIT( 0x10, IP_ACTIVE_LOW, IPT_UNKNOWN )
+	PORT_BIT( 0x20, IP_ACTIVE_LOW, IPT_UNKNOWN )
+	PORT_BIT( 0x40, IP_ACTIVE_LOW, IPT_UNKNOWN )
+	PORT_BIT( 0x80, IP_ACTIVE_LOW, IPT_UNKNOWN )
+
+	PORT_START("IN1")
+	PORT_BIT( 0x01, IP_ACTIVE_LOW, IPT_UNKNOWN )
+	PORT_BIT( 0x02, IP_ACTIVE_LOW, IPT_UNKNOWN )
+	PORT_BIT( 0x04, IP_ACTIVE_LOW, IPT_UNKNOWN )
+	PORT_BIT( 0x08, IP_ACTIVE_LOW, IPT_UNKNOWN )
+	PORT_BIT( 0x10, IP_ACTIVE_LOW, IPT_UNKNOWN )
+	PORT_BIT( 0x20, IP_ACTIVE_LOW, IPT_UNKNOWN )
+	PORT_BIT( 0x40, IP_ACTIVE_LOW, IPT_UNKNOWN )
+	PORT_BIT( 0x80, IP_ACTIVE_LOW, IPT_UNKNOWN )
+	
+	PORT_START("DSW0")
+	PORT_DIPUNKNOWN_DIPLOC(0x01, 0x01, "SW0:1")
+	PORT_DIPUNKNOWN_DIPLOC(0x02, 0x02, "SW0:2")
+	PORT_DIPUNKNOWN_DIPLOC(0x04, 0x04, "SW0:3")
+	PORT_DIPUNKNOWN_DIPLOC(0x08, 0x08, "SW0:4")
+	PORT_DIPUNKNOWN_DIPLOC(0x10, 0x10, "SW0:5")
+	PORT_DIPUNKNOWN_DIPLOC(0x20, 0x20, "SW0:6")
+	PORT_DIPUNKNOWN_DIPLOC(0x40, 0x40, "SW0:7")
+	PORT_DIPUNKNOWN_DIPLOC(0x80, 0x80, "SW0:8")
+
+	PORT_START("DSW1")
+	PORT_DIPUNKNOWN_DIPLOC(0x01, 0x01, "SW1:1")
+	PORT_DIPUNKNOWN_DIPLOC(0x02, 0x02, "SW1:2")
+	PORT_DIPUNKNOWN_DIPLOC(0x04, 0x04, "SW1:3")
+	PORT_DIPUNKNOWN_DIPLOC(0x08, 0x08, "SW1:4")
+	PORT_DIPUNKNOWN_DIPLOC(0x10, 0x10, "SW1:5")
+	PORT_DIPUNKNOWN_DIPLOC(0x20, 0x20, "SW1:6")
+	PORT_DIPUNKNOWN_DIPLOC(0x40, 0x40, "SW1:7")
+	PORT_DIPUNKNOWN_DIPLOC(0x80, 0x80, "SW1:8")
+INPUT_PORTS_END
+
+void orientalpearl_state::program_map(address_map &map)
+{
+	map(0x0000, 0xffff).rom();
+}
+
+void orientalpearl_state::io_map(address_map &map)
+{
+    map(0xfa00, 0xfa01).rw("kdc", FUNC(i8279_device::read), FUNC(i8279_device::write));
+    map(0xfb02, 0xfb03).w("psg", FUNC(ay8910_device::address_data_w));
+	map(0xfe00, 0xfe01).w("opll", FUNC(ym2413_device::write));
+	
+}
+
+void orientalpearl_state::machine_start()
+{
+}
+
+void orientalpearl_state::orientp(machine_config &config)
+{
+	/* basic machine hardware */
```

Comment:
please use C++ style comments for one-liners

Reply:
C++ style means
```cpp
// basic machine hardware
```
instead of
```cpp
/* basic machine hardware */
```